python/ray/util/client/worker.py

```diff
@@ -320,6 +340,31 @@ def _call_schedule_for_task(
                 logger.exception("Failed to deserialize {}".format(
                     ticket.error))
                 raise
+        self.total_num_tasks_scheduled += 1
+        self.total_outbound_message_size_bytes += task.ByteSize()
+        if self.total_num_tasks_scheduled > TASK_WARNING_THRESHOLD and \
+                log_once("client_communication_overhead_warning"):
+            warnings.warn(
+                f"More than {TASK_WARNING_THRESHOLD} remote tasks have been "
+                "scheduled. This can be slow on Ray Client due to "
+                "communication overhead. If you're running many fine-grained "
+                "tasks, consider running them in a single remote function."
+                "See the section on \"Too fine-grained tasks\" in the Ray "
+                "Design Patterns document for more details: "
+                f"{DESIGN_PATTERN_DOC_LINK}", UserWarning)
+        if self.total_outbound_message_size_bytes > MESSAGE_SIZE_THRESHOLD \
+                and log_once("client_communication_overhead_warning"):
+            warnings.warn(
+                "More than 10MB of messages have been created to schedule "
+                "tasks on the server. If you're running many fine-grained "
+                "tasks consider, consider running them inside a single remote "
+                "function. See the section on \"Too fine-grained tasks\" in "
+                "the Ray Design Patterns document for more details. If your "
+                "functions frequently use large objects, consider storing the "
+                "objects remotely with ray.put. An example of this is shown "
+                "in the \"Closure capture of large / unserializable object\" "
+                "section of the Ray Design Patterns document, available here: "
+                f"{DESIGN_PATTERN_DOC_LINK}", UserWarning)
```

Comment:
you need another link here, which points to the exact subsection. 
https://docs.google.com/document/d/167rnnDFIVRhHhK4mznEIemOtj63IOhtIPvSYaPgI4Fg/edit#heading=h.1afmymq455wu